src/hooks/access.ts

```diff
@@ -0,0 +1,94 @@
+import { useCallback } from 'react';
+import useSWR from 'swr';
+import { useFetcher, SwrldResult } from './things';
+import { UrlString, WebId, access } from '@inrupt/solid-client';
+
+export type AccessResult = SwrldResult & { access: access.Access; saveAccess: any };
+export type AllAccessResult = SwrldResult & {
+  allAccess: Record<WebId, access.Access>;
+};
+
+// TODO use access.Actor once https://github.com/inrupt/solid-client-js/pull/1519 is released
+export type Actor = 'agent' | 'group' | 'public';
+export const GroupActor = 'group';
+export const AgentActor = 'agent';
+export const PublicActor = 'public';
+
+export function usePublicAccess(resourceUrl: UrlString): AccessResult {
+  return useAccessFor(resourceUrl, PublicActor);
+}
+
+export function useAgentAccess(
+  resourceUrl: UrlString,
+  webId: WebId
+): AccessResult {
+  return useAccessFor(resourceUrl, AgentActor, webId);
+}
+
+export function useAgentAccessAll(
+  resourceUrl: UrlString,
+  webId: WebId
+): AllAccessResult {
+  return useAccessForAll(resourceUrl, AgentActor, webId);
+}
+
+export function useGroupAccess(
+  resourceUrl: UrlString,
+  groupId: WebId
+): AccessResult {
+  return useAccessFor(resourceUrl, GroupActor, groupId);
+}
+
+export function useGroupAccessAll(
+  resourceUrl: UrlString,
+  groupId: WebId
+): AllAccessResult {
+  return useAccessForAll(resourceUrl, GroupActor, groupId);
+}
+
+export function useAccessForAll(
+  resourceUrl: UrlString,
+  actorType: Actor,
+  actor?: WebId
+): AllAccessResult {
+  const fetcher = useFetcher(access.getAccessForAll);
+  const swr = useSWR(
+    [resourceUrl, actorType, actor],
+    fetcher
+  ) as AllAccessResult;
+  swr.allAccess = swr.data;
+  return swr;
+};
+
+export function useAccessFor(
+  resourceUrl: UrlString,
+  actorType: Actor,
+  actor?: WebId
+): AccessResult {
+  const fetcher = useFetcher(access.getAccessFor);
```

Comment:
doesn't the fetcher returned by `useFetcher` expect a single string key? I'm surprised this doesn't explode when you pass `[resourceUrl, actorType, actor]` to it in the `useSWR` in the next line...

Reply:
Lol, it does. This was the first bug I had to fix, and why I had to switch away from useFetcher.